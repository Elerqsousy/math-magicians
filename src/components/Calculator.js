import React from 'react';
import './calculator.css';

class Calculator extends React.Component {
  constructor(props) {
    super(props);
    this.state = {};
  }

  render() {
    return (
      <div className="calculator-container">
        <div className="screen"><div>0</div></div>
        <button type="button" className="btn-reset light-gray">AC</button>
        <button type="button" className="btn-change light-gray">+/-</button>
        <button type="button" className="btn-0 light-gray">0</button>
        <button type="button" className="btn-dot light-gray">.</button>
        <button type="button" className="btn-equal orange">=</button>
        <button type="button" className="btn-add orange">+</button>
        <button type="button" className="btn-subtract orange">-</button>
        <button type="button" className="btn-multiply orange">*</button>
        <button type="button" className="btn-divide orange">/</button>
        <button type="button" className="btn-percent light-gray">%</button>
        <button type="button" className="btn-1 light-gray">1</button>
        <button type="button" className="btn-2 light-gray">2</button>
        <button type="button" className="btn-3 light-gray">3</button>
        <button type="button" className="btn-4 light-gray">4</button>
        <button type="button" className="btn-5 light-gray">5</button>
        <button type="button" className="btn-6 light-gray">6</button>
        <button type="button" className="btn-7 light-gray">7</button>
        <button type="button" className="btn-8 light-gray">8</button>
        <button type="button" className="btn-9 light-gray">9</button>
      </div>
    );
  }
}

export default Calculator;
