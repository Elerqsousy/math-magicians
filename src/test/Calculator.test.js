import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import renderer from 'react-test-renderer';
import Calculator from '../components/Calculator';
import '@testing-library/jest-dom/extend-expect';

test('screen to reflect btn keys', () => {
  render(<Calculator />);
  userEvent.click(screen.getByRole('button', { name: /3/i }));
  userEvent.click(screen.getByRole('button', { name: /2/i }));

  expect(screen.getByTestId('screen')).toHaveTextContent('32');
});

it('Match the Calculator component snapshot', () => {
  const tree = renderer.create(<Calculator />).toJSON();

  expect(tree).toMatchSnapshot();
});
