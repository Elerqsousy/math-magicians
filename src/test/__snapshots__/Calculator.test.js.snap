// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Match the Calculator component snapshot 1`] = `
<div
  className="calculator-container"
>
  <div
    className="screen"
  >
    <div
      data-testid="screen"
    >
      0
    </div>
  </div>
  <button
    className="btn-reset light-gray"
    onClick={[Function]}
    onKeyDown={[Function]}
    type="button"
  >
    AC
  </button>
  <button
    className="btn-change light-gray"
    onClick={[Function]}
    onKeyDown={[Function]}
    type="button"
  >
    +/-
  </button>
  <button
    className="btn-0 light-gray"
    onClick={[Function]}
    onKeyDown={[Function]}
    type="button"
  >
    0
  </button>
  <button
    className="btn-dot light-gray"
    onClick={[Function]}
    onKeyDown={[Function]}
    type="button"
  >
    .
  </button>
  <button
    className="btn-equal orange"
    onClick={[Function]}
    onKeyDown={[Function]}
    type="button"
  >
    =
  </button>
  <button
    className="btn-add orange"
    onClick={[Function]}
    onKeyDown={[Function]}
    type="button"
  >
    +
  </button>
  <button
    className="btn-subtract orange"
    onClick={[Function]}
    onKeyDown={[Function]}
    type="button"
  >
    -
  </button>
  <button
    className="btn-multiply orange"
    onClick={[Function]}
    onKeyDown={[Function]}
    type="button"
  >
    x
  </button>
  <button
    className="btn-divide orange"
    onClick={[Function]}
    onKeyDown={[Function]}
    type="button"
  >
    รท
  </button>
  <button
    className="btn-percent light-gray"
    onClick={[Function]}
    onKeyDown={[Function]}
    type="button"
  >
    %
  </button>
  <button
    className="btn-1 light-gray"
    onClick={[Function]}
    onKeyDown={[Function]}
    type="button"
  >
    1
  </button>
  <button
    className="btn-2 light-gray"
    onClick={[Function]}
    onKeyDown={[Function]}
    type="button"
  >
    2
  </button>
  <button
    className="btn-3 light-gray"
    onClick={[Function]}
    onKeyDown={[Function]}
    type="button"
  >
    3
  </button>
  <button
    className="btn-4 light-gray"
    onClick={[Function]}
    onKeyDown={[Function]}
    type="button"
  >
    4
  </button>
  <button
    className="btn-5 light-gray"
    onClick={[Function]}
    onKeyDown={[Function]}
    type="button"
  >
    5
  </button>
  <button
    className="btn-6 light-gray"
    onClick={[Function]}
    onKeyDown={[Function]}
    type="button"
  >
    6
  </button>
  <button
    className="btn-7 light-gray"
    onClick={[Function]}
    onKeyDown={[Function]}
    type="button"
  >
    7
  </button>
  <button
    className="btn-8 light-gray"
    onClick={[Function]}
    onKeyDown={[Function]}
    type="button"
  >
    8
  </button>
  <button
    className="btn-9 light-gray"
    onClick={[Function]}
    onKeyDown={[Function]}
    type="button"
  >
    9
  </button>
</div>
`;
